tx-pool/src/process.rs

```diff
@@ -652,117 +654,182 @@ impl TxPoolService {
-        self.chunk.read().await.is_full()
+    pub(crate) async fn run_verify_tx(
+        &mut self,
+        entry: Entry,
+        command_rx: &mut watch::Receiver<ChunkCommand>,
+    ) -> Option<(Result<Completed, Reject>, Arc<Snapshot>)> {
+        let Entry { tx, remote } = entry;
+        let tx_hash = tx.hash();
+
+        let (ret, snapshot) = self.pre_check(&tx).await;
+        let (tip_hash, rtx, status, fee, tx_size) = try_or_return_with_snapshot!(ret, snapshot);
+        let cached = self.fetch_tx_verify_cache(&tx_hash).await;
+        let tip_header = snapshot.tip_header();
+        let consensus = snapshot.cloned_consensus();
+
+        let tx_env = Arc::new(TxVerifyEnv::new_submit(tip_header));
+
+        if let Some(ref completed) = cached {
+            let ret = TimeRelativeTransactionVerifier::new(
+                Arc::clone(&rtx),
+                Arc::clone(&consensus),
+                snapshot.as_data_loader(),
+                Arc::clone(&tx_env),
+            )
+            .verify()
+            .map(|_| *completed)
+            .map_err(Reject::Verification);
+            let completed = try_or_return_with_snapshot!(ret, snapshot);
+
+            let entry = TxEntry::new(rtx, completed.cycles, fee, tx_size);
+            let (ret, snapshot) = self.submit_entry(tip_hash, entry, status).await;
+            try_or_return_with_snapshot!(ret, snapshot);
+            return Some((Ok(completed), snapshot));
+        }
+
+        let cloned_snapshot = Arc::clone(&snapshot);
+        let data_loader = cloned_snapshot.as_data_loader();
+        let ret = ContextualWithoutScriptTransactionVerifier::new(
+            Arc::clone(&rtx),
+            Arc::clone(&consensus),
+            data_loader.clone(),
+            Arc::clone(&tx_env),
+        )
+        .verify()
+        .and_then(|result| {
+            DaoScriptSizeVerifier::new(
+                Arc::clone(&rtx),
+                Arc::clone(&consensus),
+                data_loader.clone(),
+            )
+            .verify()?;
+            Ok(result)
+        })
+        .map_err(Reject::Verification);
+        let fee = try_or_return_with_snapshot!(ret, snapshot);
+
+        let max_cycles = if let Some((declared_cycle, _peer)) = remote {
+            declared_cycle
+        } else {
+            consensus.max_block_cycles()
+        };
+
+        let ret = {
+            let script_verifier =
+                ScriptVerifier::new(Arc::clone(&rtx), data_loader, consensus, tx_env);
+            script_verifier
+                .resumable_verify_with_signal(max_cycles, command_rx)
+                .await
+                .map_err(Reject::Verification)
+        };
+
+        let state = try_or_return_with_snapshot!(ret, snapshot);
+        let completed: Completed = match state {
+            VerifyResult::Completed(cycles) => Completed { cycles, fee },
+            _ => {
+                panic!("not expected");
+            }
+        };
```

Comment:
Can we enhance this code block so that `resumable_verify_with_signal` can return `Ok(cycles, fee)` or `Err(err)`, allowing us to avoid using `panic!` in this context?

Reply:
`fee` is not returned by `resumable_verify_with_signal`, the root cause is I'm still using `VerifyResult`, which has `Suspend`.